x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/form/validations.ts

```diff
@@ -56,33 +56,29 @@ export const ROLLOVER_EMPTY_VALIDATION = 'ROLLOVER_EMPTY_VALIDATION';
  * This validator checks that and updates form values by setting errors states imperatively to
  * indicate this error state.
  */
-export const rolloverThresholdsValidator: ValidationFunc = ({ form }) => {
+export const rolloverThresholdsValidator: ValidationFunc = ({ form, path }) => {
   const fields = form.getFields();
   if (
     !(
-      fields[ROLLOVER_FORM_PATHS.maxAge].value ||
-      fields[ROLLOVER_FORM_PATHS.maxDocs].value ||
-      fields[ROLLOVER_FORM_PATHS.maxSize].value
+      fields[ROLLOVER_FORM_PATHS.maxAge]?.value ||
+      fields[ROLLOVER_FORM_PATHS.maxDocs]?.value ||
+      fields[ROLLOVER_FORM_PATHS.maxSize]?.value
     )
   ) {
-    fields[ROLLOVER_FORM_PATHS.maxAge].setErrors([
-      {
-        validationType: ROLLOVER_EMPTY_VALIDATION,
-        message: i18nTexts.editPolicy.errors.maximumAgeRequiredMessage,
-      },
-    ]);
-    fields[ROLLOVER_FORM_PATHS.maxDocs].setErrors([
-      {
-        validationType: ROLLOVER_EMPTY_VALIDATION,
-        message: i18nTexts.editPolicy.errors.maximumDocumentsRequiredMessage,
-      },
-    ]);
-    fields[ROLLOVER_FORM_PATHS.maxSize].setErrors([
-      {
-        validationType: ROLLOVER_EMPTY_VALIDATION,
-        message: i18nTexts.editPolicy.errors.maximumSizeRequiredMessage,
-      },
-    ]);
+    return path === ROLLOVER_FORM_PATHS.maxAge
```

Comment:
nit: might be a little easier to read using `if...else` instead of a nested ternary.

Reply:
A switch/case could also work.

Reply:
@cjcenizal in this case the switch statement would look something like:

```ts
    switch (true) {
      case 1 === 1:
        return 'ok';
      case 2 === 2:
        return 'ok then';
      default:
        return 'not ok';
    }
```

Just curious to hear thoughts on that pattern where we have `true` as the deciding condition and predicate expressions per case. IMO it might be less easy to grok for the majority of readers because it is less common - WDYT?

Reply:
My take is it does look pretty neat though!

Reply:
@jloleysens I was thinking something more like this unless I'm missing something:

```js
switch (path) {
  case ROLLOVER_FORM_PATHS.maxAge:
    return {
      code: ROLLOVER_EMPTY_VALIDATION,
      message: i18nTexts.editPolicy.errors.maximumAgeRequiredMessage,
    };

  case ROLLOVER_FORM_PATHS.maxDocs:
    return {
      code: ROLLOVER_EMPTY_VALIDATION,
      message: i18nTexts.editPolicy.errors.maximumDocumentsRequiredMessage,
    };

  default:
    return {
      code: ROLLOVER_EMPTY_VALIDATION,
      message: i18nTexts.editPolicy.errors.maximumSizeRequiredMessage,
    };
}
```

Reply:
Ah OK, I had a different idea. Yours is simpler!